validator/client/src/main/java/tech/pegasys/teku/validator/client/duties/attestations/AttestationProductionDuty.java

```diff
@@ -144,7 +152,12 @@ private SafeFuture<ProductionResult<Attestation>> signAttestationForValidatorInC
                 maybeUnsignedAttestation
                     .map(
                         attestationData ->
-                            signAttestationForValidator(slot, forkInfo, attestationData, validator))
+                            validatorDutyMetrics.record(
+                                () ->
+                                    signAttestationForValidator(
+                                        slot, forkInfo, attestationData, validator),
+                                this,
+                                Step.SIGN))
```

Comment:
I think we should do the signing inside `signAttestationForValidator` (at line 185).

The reason being: into that method, there is a `checkArgument` that checks if we are even going to try signing it. We don't want it to mark the time of the sign operation when this check fails, because it has nothing to do with signing and it this could artificially lower the average signing time.

Reply:
Another option is moving the check out of that method and having a validateAttestation or something.